src/RA_Dlg_Achievement.cpp

```diff
@@ -1190,20 +1199,34 @@ void Dlg_Achievements::OnEditData(size_t nItem, Column nColumn, const std::strin
     if (nItem >= m_lbxData.size())
         return;
 
-    m_lbxData[nItem][nColumn] = sNewData;
+    m_lbxData.at(nItem).at(ra::etoi(nColumn)) = sNewData;
 
     HWND hList = GetDlgItem(m_hAchievementsDlg, IDC_RA_LISTACHIEVEMENTS);
     if (hList != nullptr)
     {
-        LV_ITEM item;
-        ZeroMemory(&item, sizeof(item));
-
-        item.mask = LVIF_TEXT;
-        item.iItem = nItem;
-        item.iSubItem = nColumn;
-        item.cchTextMax = 256;
-        ra::tstring sStr = NativeStr(m_lbxData[nItem][nColumn]);	//	scoped cache
-        item.pszText = sStr.data();
+        ra::tstring sStr{ NativeStr(LbxDataAt(nItem, nColumn)) }; // scoped cache
+        LV_ITEM item
+        {
+            item.mask       = ra::to_unsigned(LVIF_TEXT),
+            item.iItem      = nItem,
+            item.iSubItem   = ra::narrow_cast<int>(ra::to_signed(ra::etoi(nColumn))),
+            item.state      = 0U,
+            item.stateMask  = 0U,
+            item.pszText    = sStr.data(),
+            item.cchTextMax = 256,
+            item.iImage     = 0,
+            item.lParam     = 0L,
+            item.iIndent    = 0,
+            item.iGroupId   = 0,
+            item.cColumns   = 0U,
+            item.puColumns  = nullptr,
+#if _WIN32_WINNT >= _WIN32_WINNT_VISTA
+            item.piColFmt   = nullptr, // still needs to be initialized to prevent corruption
+            item.iGroup     = 0
+#endif /* _WIN32_WINNT >= _WIN32_WINNT_VISTA */
```

Comment:
`ZeroMemory` seemed cleaner. What if you missed one? Or someone in the future accidentally removes one? And eliminating the #if is desirable.

Reply:
Or if you're completely against `ZeroMemory` for whatever reason, use the default initializer:
```
LV_ITEM item{}
item.mask = ...
```